File: BergerMsfaApi/ClientApp/src/app/Shared/Entity/Menu/menu-activity.ts
import { Menu } from './menu.model';
import { MenuActivityPermission } from './menu-activity-permission.model';

export class MenuActivity {
    public id: number;
    public status: number;
    public statusText: string;
    public name: string;
    public activityCode: string;
    public menuId: number;
    public menu: Menu;
    public menuName: string;
    public menuCode: string;
    public menuActivityPermission : MenuActivityPermission;

    // constructor() {
    //     this.id = 0;
    //     this.status = 1;
    //     this.statusText = "";
    //     this.name = "";
    //     this.activityCode = "";
    //     this.menuId = 0;
    //     this.menu = new Menu();
    //     this.menuName = "";
    // }
    
}

File: Berger.Data/MsfaEntity/Menus/MenuActivity.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Berger.Data.Attributes;
using Berger.Data.Common;

namespace Berger.Data.MsfaEntity.Menus
{
    public class MenuActivity : AuditableEntity<int>
    {
        [UniqueKey()]
        [Column(Order = 1)]
        [Required]
        [StringLength(256, MinimumLength = 1)]
        public string Name { get; set; }

        [UniqueKey()]
        [Column(Order = 2)]
        public string ActivityCode { get; set; }

        [Column(Order = 3)]
        [ForeignKey("MenuId")]
        public int MenuId { get; set; }
        
        public Menu Menu { get; set; }
        public List<MenuActivityPermission> Permissions { get; set; }

    }
}
